pulp_smash/tests/rpm/api_v2/test_upload_publish.py

```diff
@@ -367,99 +243,3 @@ def verify_repo_download(self, repo):
-        client = api.Client(cls.cfg, api.json_handler)
-        body = gen_repo()
-        body['importer_config']['feed'] = RPM_UNSIGNED_FEED_URL
-        body['distributors'] = [gen_distributor()]
-        repo = client.post(REPOSITORY_PATH, body)
-        cls.resources.add(repo['_href'])
-
-        # Sync content into the repository, and give it an erratum.
-        utils.sync_repo(cls.cfg, repo['_href'])
-        utils.upload_import_erratum(cls.cfg, cls.erratum, repo['_href'])
-        repo = client.get(repo['_href'], params={'details': True})
-
-        # Publish the repository, and fetch and parse updateinfo.xml
-        utils.publish_repo(cls.cfg, repo)
-        cls.updateinfo = (
-            get_repodata(cls.cfg, repo['distributors'][0], 'updateinfo')
-        )
-
-    def test_updateinfo_root_tag(self):
-        """Assert ``updateinfo.xml`` has a root element named ``updates``."""
-        self.assertEqual(self.updateinfo.tag, 'updates')
-
-    def test_updates_are_present(self):
-        """Assert at least one update is present in ``updateinfo.xml``."""
-        n_updates = len(self.updateinfo.findall('update'))
-        self.assertGreater(n_updates, 0)
-
-    def test_update_is_correct(self):
-        """Assert the uploaded erratum is in ``updateinfo.xml`` and correct.
-
-        Specificially, this method does the following:
-
-        1. Select the "update" element in ``updateinfo.xml`` corresponding to
-           the uploaded erratum.
-        2. Build our own erratum from the selected "update" element.
-        3. Compare the uploaded erratum to the generated erratum.
-
-        .. WARNING:: This test may be erroneous. See
-            :func:`parse_updateinfo_update`.
-        """
-        # Find our erratum in ``updateinfo.xml``.
-        updates = [
-            update for update in self.updateinfo.findall('update')
-            if update.find('id').text == self.erratum['id']
-        ]
-        self.assertEqual(len(updates), 1)
-
-        # The erratum we uploaded will be different from the erratum that Pulp
-        # published. Here, we try to erase the differences between the two, by
-        # modifying the uploaded erratum.
-        old_erratum = self.erratum.copy()  # dont modify original
-        if selectors.bug_is_untestable(2021, self.cfg.version):
-            del old_erratum['release']
-        if selectors.bug_is_testable(2042, self.cfg.version):
-            for package_list in old_erratum['pkglist']:
-                for package in package_list['packages']:
-                    # ['md5', '…', 'sha256', '…'] → ['sha256': '…']
-                    i = package['sum'].index('sha256')
-                    package['sum'] = package['sum'][i:i + 2]
-        else:
-            for package_list in old_erratum['pkglist']:
-                for package in package_list['packages']:
-                    del package['sum']
-        new_erratum = parse_updateinfo_update(updates[0])
```

Comment:
The idea here was to take the erratum uploaded to Pulp, transform it to the form we expect Pulp to publish, and then compare the transformed erratum to what Pulp actually publishes. It works, but the needed code is ugly as sin. It turns out that it's easier to just make dumb, repetitive assertions like what's done with the `verify_*` methods.